### PR TITLE
Add mmft 0.2.1

### PR DESCRIPTION
**Pull request by Euphrasiologist**
Initial add of mmft to conda.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/mmft/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -ex
+
+RUST_BACKTRACE=full
+
+if [ "$(uname)" == "Darwin" ]; then
+    # apparently the HOME variable isn't set correctly, and circle ci output indicates the following as the home directory
+    export HOME="/Users/distiller"
+    export HOME=`pwd`
```
Comment (martin-g) [inline recipes/mmft/build.sh:9]: What's the idea of exporting HOME twice?
Only the second will survive.

File: recipes/mmft/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    echo "HOME is $HOME"
+
+    # according to https://github.com/rust-lang/cargo/issues/2422#issuecomment-198458960 removing circle ci default configuration solves cargo trouble downloading crates
+    #git config --global --unset url.ssh://git@github.com.insteadOf
+fi
+
+cargo install -v --locked --root "$PREFIX" --path .
```
Comment (martin-g) [inline recipes/mmft/build.sh:16]:
```suggestion
cargo install -v --locked --root "$PREFIX" --path . --no-track
```

File: recipes/mmft/build.sh (hunk 1 above)
Comment (martin-g) [reply recipes/mmft/build.sh:9]: Is it really needed ?
How does it fail without this code ?

Comment (Euphrasiologist) [reply recipes/mmft/build.sh:7]: I've removed so we will see.

**Comment by Euphrasiologist** [comment]
@BiocondaBot please add label

File: recipes/mmft/build.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+RUST_BACKTRACE=full
+
+if [ "$(uname)" == "Darwin" ]; then
+    export HOME=`pwd`
+    echo "HOME is $HOME"
+
+    # according to https://github.com/rust-lang/cargo/issues/2422#issuecomment-198458960 removing circle ci default configuration solves cargo trouble downloading crates
+    #git config --global --unset url.ssh://git@github.com.insteadOf
+fi
```
Comment (mbhall88) [inline recipes/mmft/build.sh:12]: This is required anymore. See some recent Rust packages

* https://bioconda.github.io/contributor/guidelines.html#rust
* https://github.com/bioconda/bioconda-recipes/blob/master/recipes/lrge/build.sh

File: recipes/mmft/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+test:
+  commands:
+    - mmft --help
+    - mmft sample --help
+
+about:
+  home: https://github.com/ARU-life-sciences/mmft
+  license: MIT
+  summary: Max's minimal fasta toolkit
+  license_file: LICENSE
```
Comment (mbhall88) [inline recipes/mmft/meta.yaml:31]: Please see the Rust package guidelines for how to include the license of dependencies for Rust packages

https://bioconda.github.io/contributor/guidelines.html#rust

**Review by mbhall88 — CHANGES_REQUESTED** [review]

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
@mbhall88 I've applied your suggestions!

File: recipes/mmft/build.sh (hunk 3 above)
Comment (Euphrasiologist) [reply recipes/mmft/build.sh]: @martin-g beat me to it!

**Comment by Euphrasiologist** [comment]
Is there anything else to do here?

**Review by mbhall88 — APPROVED** [review]
Thanks both. Looks good

**Pull request merged**